backend/apps/admin/src/problem/problem.service.ts

```diff
@@ -0,0 +1,211 @@
+      const parsedTag = problemCreateInput.problemTag.map(
+        (value): TagCreateManyInput => {
+          return {
+            name: value
+          }
+        }
+      )
+      await this.prisma.tag.createMany({
+        data: parsedTag,
+        skipDuplicates: true
+      })
+      const _tagsId = await this.prisma.tag.findMany({
+        select: { id: true },
+        where: {
+          OR: parsedTag
+        }
+      })
+      tagsId = _tagsId.map((value) => {
+        return {
+          tagId: value.id
+        }
+      })
+    }
+
+    const {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      groupId: _1,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      createdById: _3,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      problemTestcase: _4,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      problemTag: _5,
+      ..._data
+    } = problemCreateInput
+
+    try {
+      const createdProblem = await this.prisma.problem.create({
+        data: {
+          ..._data,
+          group: group,
+          createdBy: createdBy,
+          problemTestcase: problemTestcase,
+          problemTag: {
+            create: tagsId
+          }
+        }
+      })
+      return `${createdProblem.title} has been successfully created with ID numbered ${createdProblem.id}!`
+    } catch {
+      throw new BadRequestException()
+    }
+  }
+
+  async findAll(getGroupProblemsInput: GetGroupProblemsDto) {
+    const whereOptions: ProblemWhereInput = {
+      groupId: { equals: getGroupProblemsInput.groupId }
+    }
+
+    if (isDefined(getGroupProblemsInput.createdById)) {
+      whereOptions.createdById = { equals: getGroupProblemsInput.createdById }
+    }
+
+    if (isDefined(getGroupProblemsInput.difficulty)) {
+      whereOptions.difficulty = {
+        equals: getGroupProblemsInput.difficulty
+      }
+    }
+
+    if (isDefined(getGroupProblemsInput.languages)) {
+      whereOptions.languages = { hasSome: getGroupProblemsInput.languages }
+    }
+
+    return await this.prisma.problem.findMany({
+      where: {
+        ...whereOptions
+      },
+      select: this.problemSelectOption,
+      cursor: {
+        id: getGroupProblemsInput.cursor < 1 ? 1 : getGroupProblemsInput.cursor
+      },
+      skip: getGroupProblemsInput.cursor < 1 ? 0 : 1,
+      take: getGroupProblemsInput.take
+    })
+  }
+
+  async findOne(getGroupProblemInput: GetGroupProblemDto) {
+    try {
+      return await this.prisma.problem.findUniqueOrThrow({
+        where: {
+          id: getGroupProblemInput.problemId
+        },
+        select: this.problemSelectOption
+      })
+    } catch {
+      throw new NotFoundException()
+    }
+  }
+
+  async update(updateProblemInput: UpdateProblem) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { problemId: _a, difficulty: _b, ..._data } = updateProblemInput
```

Comment:
여기서 쓰신 문법은 [구조 분해 할당](https://basarat.gitbook.io/typescript/future-javascript/destructuring)이라고 불리는 문법이에요! 우항에 주어진 object의 property를 분해해서 좌항의 변수들에 나누어 할당해주는 문법입니다. 그러니까
```ts
const updateProblemInput = {
  problemId: 1,
  difficulty: 2,
  rest1: 3,
  rest2: 4
}
const { problemId: _a, difficulty: _b, ..._data } = updateProblemInput /*
const _a = 1
const _b = 2
const _data = {
  rest1: 3,
  rest2: 4
} */
```
쓰신 코드는 제일 아래의 주석과 똑같은 거예요! 그래서 처음부터 아래처럼 할당하시면, 뒤에서 problemId나 difficulty를 굳이 다시 만들어줄 필요가 없습니다. 그리고 difficulty는 처음부터 required 필드로 설정하는 게 좋을 것 같아요!
```suggestion
const { problemId, difficulty, ...problem } = updateProblemInput
```